src/components/menu/MenuTopLevelControlled.tsx

```diff
@@ -0,0 +1,68 @@
+// (C) 2007-2018 GoodData Corporation
+import * as React from 'react';
+import { OpenAction, IMenuPositionConfig } from './MenuSharedTypes';
+import MenuOpener from './menuOpener/MenuOpener';
+
+export interface IMenuTopLevelControlledProps extends Partial<IMenuPositionConfig> {
+    opened: boolean;
+    openAction?: OpenAction;
+    closeOnScroll: boolean;
+    portalTarget: Element;
+    onOpenedChange: (opened: boolean) => void;
+    toggler: React.ReactNode;
+    children: React.ReactNode;
+}
+
+export default class MenuTopLevelControlled extends React.Component<IMenuTopLevelControlledProps> {
```

Comment:
I would say `ControlledMenu` is more suitable name for component

Reply:
Renamed